app/helpers/govspeak_helper.rb

```diff
@@ -63,7 +63,9 @@ def prepare_images(images)
   end
 
   def prepare_attachments(attachments, alternative_format_contact_email)
-    attachments.map do |attachment|
+    attachments
+      .select { |attachment| !attachment.file? || attachment.attachment_data.all_asset_variants_uploaded? }
```

Comment:
could make the calling code use `prepare_attachments(item.attachments_ready_for_publishing...)`

Reply:
As discussed, this is better for now, as we the alternative would force us to write excessive tests or refactor the abstractions in Presenters to better represent the requirements - not the right time for that.